docs/index.html

```diff
@@ -31,19 +31,76 @@
   <meta name="msapplication-TileColor" content="#da532c">
   <meta name="msapplication-TileImage" content="./logos/mstile-144x144.png">
   <meta name="theme-color" content="#ffffff">
-  <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/devicons/devicon/devicon.min.css">
   <link rel="stylesheet" href="./assets/css/style.css">
   <meta property="og:title" content="Devicon" />
   <meta property="og:image" content="https://raw.githubusercontent.com/devicons/devicon/master/icons/devicon/devicon-original-wordmark.svg" />
   <meta property="og:url" content="https://devicon.dev" />
   <meta property="og:type" content="website" />
   <meta property="og:locale" content="en_US" />
   <meta property="og:description" content="Devicon aims to gather all logos representing development languages and tools in one font. 🚀" />
+  <script>
+    /////////////////////////////
+    // Basic helper functions //
+    /////////////////////////////
+    function addCss(href) {
+      var link = document.createElement('link');
+
+      link.rel = 'stylesheet';
+      link.type = 'text/css';
+      link.href = `${href}`;
+
+      document.body.appendChild(link);
+      document.head.appendChild(link);
+    }
+
+    function addImage(src, id, img_class) {
+      var img = document.createElement("img");
+
+      img.src = src;
+      img.alt = "{{selectedIcon.name}}-{{svgVersion}}";
+      img.classList.add(img_class);
+
+      document.getElementById(id).appendChild(img);
+    }
+
+    // Unused. May be useful in the future.
+    function addElement(content, id) {
+      document.getElementById(id).innerHTML = content;
+    }
+
+    ///////////////////////
+    // Path management //
+    /////////////////////
+    var path = '/devicon.min.css';
+    var baseUrl
+  
+    var xhr = new XMLHttpRequest();
+    xhr.open('HEAD', path, false);
+    xhr.send();
+
+    // Make sure the file exists, otherwise, use latest CDN link.
+    if (xhr.status == "404") {
+      baseUrl = `https://cdn.jsdelivr.net/gh/devicons/devicon@latest/`;
+    } else {
+      baseUrl = `/`;
+    }
+
+    // Temporary solution.
+    var counter = 0
+    var timer = setInterval(function() {
+          ++counter;
+
+          addCss(`${baseUrl}devicon.min.css`);
+          addImage(`${baseUrl}icons/{{selectedIcon.name}}/{{selectedIcon.name}}-{{svgVersion}}.svg`, "svg-list", 'svgLogo');
+
+          if (counter == 1) {
+            clearInterval(timer);
+          }
+        }, 0);
```

Comment:
I'm a bit confused. Could you tell me what this code does?

Reply:
This will add the `devicon.min.css` stylesheet in the HTML head, however, we must verify whether the website is being built locally or using the NPM provided files, that's why this is done on this script.
The image addition is adding the selected icon, but we must determine whether local or NPM files should be used (see the `svgLogo` ID in one of the HTML `div`s for more information).

The counter interval would usually be useful to add multiple instances of those (we just need one here, though). However, when I tried that without the interval, it didn't add them properly or work, so I used that as a workaround.

Reply:
I see. The need for a timer seems odd, any idea why it works with the timer, but not without it?

Reply:
> I see. The need for a timer seems odd, any idea why it works with the timer, but not without it?

Nope, I don't have a clue why that happens. 🤔

Reply:
Oh well, if it works... It works 😅

Reply:
Actually, can we move this to the index.js file instead? Then we keep the script separate from the html, like before.

If you allow me to edit your branch I can probably work on it tomorrow :)